claim/validations.py

```diff
@@ -994,7 +994,9 @@ def process_dedrem(claim, audit_user_id=-1, is_process=False):
     ).values("policy_id", "product_id")
     if items_query.count() == 0 and services_query.count() == 0:
         logger.warning(f"claim {claim.uuid} did not have any item or service to valuate.")
-    for policy_product in items_query.union(services_query, all=True):
+    items_services_union_list = list(items_query.union(services_query, all=True))
+    items_services_union_no_duplicates = [dict(t) for t in {tuple(d.items()) for d in items_services_union_list}]
```

Comment:
This operation is also very expensive. For removing duplicates directly on the QuerySet object you can use .distinct() operator. 

Side note: This operation has a lot of complexity. First thing is double iteration, another is multple type casts. 
In django there are plenty of methods and parametrization of for returning values (I think in your case values_list with flat=True would be desired, then it can be casted to the set which will remove duplicates, no need for additional list cast). Remember that set is unordered which might cause issues in some cases.

Reply:
distinct() didn't work here
I just removed that part because it turns out duplicates in query weren't the root of the issue